Dockerfile

```diff
@@ -1,96 +1,191 @@
+    -DBUILD_TESTING=OFF \
+    -DRUN_LONG_TESTS=OFF \
+    -DRUN_SHORT_TESTS=OFF ../ANTs &&\
+    make -j 2
+WORKDIR /ants_build/ANTS-build
+RUN make install
+ENV ANTSPATH=/opt/ANTs/bin/
 ENV PATH=$PATH:$ANTSPATH
 
 WORKDIR /
-RUN apt-get -y install unzip
-RUN git clone https://github.com/MRtrix3/mrtrix3.git
+RUN apt-get -y install unzip qt5-default &&\
+    git clone https://github.com/MRtrix3/mrtrix3.git
 WORKDIR /mrtrix3
-RUN git fetch --tags
-RUN git checkout tags/3.0_RC3 -b 3.0_RC3
-RUN ./configure
-RUN NUMBER_OF_PROCESSORS=8 ./build
+RUN git fetch --tags &&\
+    git checkout tags/3.0_RC3 -b 3.0_RC3 &&\
+    ./configure &&\
+    NUMBER_OF_PROCESSORS=8 ./build
 ENV PATH=/mrtrix3/bin:$PATH
+
+RUN DEBIAN_FRONTEND=noninteractive apt-get -y install libsm6 libxext6 libxrender-dev python3-pip python3.7 libtool autoconf pkg-config &&\
+    update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.7 1 &&\
+    update-alternatives --config python3 &&\
+    update-alternatives  --set python3 /usr/bin/python3.7 &&\
+    python3.7 -m pip install pip &&\
+    pip3 install --upgrade pip &&\
+    apt-get -y install python3-lxml python3-six python3.7-dev python3.7-tk
+
+ENV MESA=mesa-19.0.8.tar.gz
+ENV VTK=VTK-8.2.0.tar.gz
+
+RUN rm -rf /VTK-src && mkdir /VTK-src
+RUN rm -rf /VTK-build && mkdir /VTK-build
+RUN rm -rf /mesa-src && mkdir /mesa-src
+
+COPY $VTK /VTK-src
+COPY $MESA /mesa-src
+
+WORKDIR /VTK-src
+RUN tar zxvf $VTK && rm $VTK
+
+WORKDIR /mesa-src
+RUN tar zxvf $MESA && rm $MESA
+
+WORKDIR /mesa-src/mesa-19.0.8
+RUN apt install -y xorg-dev\
+                   llvm-7 llvm-7-dev llvm-7-runtime
+RUN ./configure --prefix=/usr/ --enable-autotools                   \
+                  --enable-opengl --disable-gles1 --disable-gles2   \
+                  --disable-va --disable-xvmc --disable-vdpau       \
+                  --enable-shared-glapi                             \
+                  --disable-texture-float                           \
+                  --enable-gallium-llvm --enable-llvm-shared-libs   \
+                  --with-gallium-drivers=swrast,swr                 \
+                  --disable-dri --with-dri-drivers=                 \
+                  --disable-egl --with-egl-platforms= --disable-gbm \
+                  --disable-glx                                     \
+                  --disable-osmesa --enable-gallium-osmesa --with-llvm-prefix=/usr/lib/llvm-7 &&\
+    make -j 10 &&\
+    make install
+
+
+RUN update-alternatives --config python3 &&\
+    update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.7 1 &&\
+    update-alternatives  --set python3 /usr/bin/python3.7
+
+ENV PYTHON_INCLUDE_DIR=/usr/include/python3.7
+ENV PYTHON_LIBS=/usr/lib/python3.7/config-3.7m-x86_64-linux-gnu/libpython3.7.so
+ENV PYTHON_LIBRARY=/usr/lib/python3.7/config-3.7m-x86_64-linux-gnu/libpython3.7.so
+RUN pip3 install -U setuptools
+
+RUN mv /usr/bin/python2 /usr/bin/bk_python2
```

Comment:
Is it not easiest to directly use python3.7 as python and just ignore python2 ?

Reply:
Tried but didnt want to break anything I think this is a very small detail and won't change anything.